arch/riscv/mach-dpu/ssi_flash.c

```diff
@@ -49,27 +49,37 @@ void __dpu_ssi_flash_boot(void *boot, uint32_t addr, uint32_t size)
 {
 	int i;
 	uint8_t *dst = boot;
-	void (*boot_entry)(void) = boot;
 
-	for (i = 0; i < size; i++, addr++)
+	for (i = 0; i < size; i++, addr++) {
+		if (i == 0)
+			printf("ssi copy\n");
+
+		if ((i % 10000) == 0)
+			printf("%d ", i);
+
 		dst[i] = dpu_ssi_flash_read(addr);
-	boot_entry();
+
+		if (i == 0)
+			printf("\nfirst coppied\n");
+	}
+
+	printf("copied from ssi\n");
```

Comment:
I doubt why boot_entry mechanism cannot be applied here.
And why printf(s) are used here.